### DISCUSSION TITLE
Is there a way to set the default priority fee?

### DISCUSSION THREAD

**Discussion by levity** [ConsenSys-archive/ganache#3507] [Q&A]
When I start Ganache with the `--miner.defaultGasPrice` option, its value ends up being used in Metamask for both the base fee and the priority fee. Is there a way to set these separately?

**Comment by MicaiahReid** [comment]
Currently, if Ganache is auto-filling transaction defaults, the `miner.defaultGasPrice` option is only used to set `gasPrice`, a field which only exists on legacy and Type 1 transactions. Ganache defaults the `maxPriorityFeePerGas` to one GWEI (we have [an issue](https://github.com/trufflesuite/ganache/issues/2097) to make this more robust) and the `maxFeePerGas` to the max possible `baseFeePerGas` three blocks from the current `latest` block.

All that to say, if you're seeing that the `miner.defaultGasPrice` option is being used for both the `maxFeePerGas` and `maxPriorityFeePerGas` when connecting with MetaMask, I think that MetaMask is auto-filling these values, not Ganache. I'm not familiar with MetaMask internals, but I'd imagine they are calling out to the `eth_gasPrice` RPC method (which returns `miner.defaultGasPrice`) to help calculate their auto-fill values, but I agree that it seems strange for them to use that value for the priority fee. You might want to open an issue with MetaMask to find out.

As an additional note, we do have [an issue](https://github.com/trufflesuite/ganache/issues/3358) to make Ganache's `miner.defaultGasPrice` field work for auto-filling Type 2 transactions, but I don't think that will resolve your issue, as it seems that MetaMask is doing the auto-filling.

Hope that helps!
